__tests__/app.test.js

```diff
@@ -23,4 +23,58 @@ describe('GET /api', () => {
           expect(body.endpoints).toEqual(endpoints);
         });
     });
-  });
\ No newline at end of file
+  });
+
+  describe("invalid endpoint", () => {
+    test("responds with a 404 status and an arror message ehen given an endpoint that does nor exist", () => {
+      return request(app)
+      .get("/api/not-an-endpoint")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.message).toBe('Not Found')
+      })
+    })
+  })
+
+  describe('GET /api/topics', () => {
+    test('responds with status 200 and an array containing data of all topics', () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body.topics)).toBe(true);
+          expect(body.topics.length).toBeGreaterThan(0);
+  
+          expect(body.topics[0]).toEqual({
+            description: 'The man, the Mitch, the legend',
+            slug: 'mitch'
+          });
+  
+          body.topics.forEach(topic => {
+            expect(topic).toEqual(
+              expect.objectContaining({
+                slug: expect.any(String),
+                description: expect.any(String)
+              })
+            );
+          });
+        });
+    });
+  });
+
+  describe('status 500', () => {
+    test("responds with status 500 when database connection fails", () => {
+      jest.spyOn(db, 'query').mockImplementation(() => {
+        throw new Error('Database connection error');
+      });
+  
+      return request(app)
+        .get("/api/topics")
+        .expect(500)
+        .then(response => {
+          expect(response.body.msg).toBe("Internal Server Error");
+        });
```

Comment:
I don't often see tests for a `500` internal server error, but I really like this! What a fantastic use of `mockImplementation` 🏆